### PR TITLE
feat: add windows installer

### LINKED ISSUE #302: Help validate Windows installer publishing
#### Reason/Context

[This](https://github.com/asyncapi/cli/pull/187) PR introduced automation for publishing of MacOS `pkg` file. It was also designed in the way that it should be easy to plug in generation of installer files for Windows and Linux.

**This issue is for folks that work on Windows machine** and can do some local testing to check if all works as expected

#### Description

* First make sure installer generation works:
  * run `npm run pack:windows`
  * installer should be produced in `dist` directory. Try to install CLI using it
* Then open a PR with the following changes
  * Uncomment this part in the artifact publishing job: https://github.com/asyncapi/cli/blob/master/.github/workflows/upload-release-assets.yml#L22-L25 and modify as much as needed
  * I don't know what will be the name of the generated installer, so you will probably have to refactor existing script for MacOS installer renaming -> https://github.com/asyncapi/cli/blob/master/scripts/macosPackRename.js
  * Update Readme instructions on how to quickly install CLI with new installer -> https://github.com/asyncapi/cli#installation

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**

This PR is adding windows installer to our generated assets in the release, `pack:windows` generates installer application for both `x64` and `x86` architechture windows.

- `asyncapi-v0.38.0-893aaaa-x64.exe` renamed to `asyncapi-x64.exe`
- `asyncapi-v0.38.0-893aaaa-x86.exe` renamed to `asyncapi-x86.exe`

**Related issue(s)**

Resolves #302

### CODE REVIEW

**Comment by Souvikns** [comment]
@derberg, ready for review

File: docs/installation.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ npm install -g @asyncapi/cli
 ## Mac
 There are two ways to install the AsyncAPI CLI on your macOS: using the `brew` package manager or `pkg` files.
 
+### Windows
```
Comment (derberg) [inline docs/installation.md:32]: wrong location of section, Windows should not be under Mac

also please provide direct links to 64 and 32 bit files like we do for other distros -> https://github.com/asyncapi/cli/releases/latest/download/asyncapi.x64.pkg

these can be "predicted"

File: .github/workflows/upload-release-assets.yml [hunk 2]
```diff
@@ -23,10 +23,10 @@ jobs:
...
             dist_folder: tar
             extension: tar.gz
-          # - os: windows-latest
-          #   npm_script: pack:windows
-          #   dist_folder: win
-          #   extension: exe
+          - os: windows-latest
+            npm_script: pack:windows
+            dist_folder: win
+            extension: exe
```
[the remark this answers is not in the document]
Comment (Souvikns) [reply .github/workflows/upload-release-assets.yml:29]: I think I get what you are saying, for windows `.x64.exe` and `.x86.exe` should be the extensions, and we update the CI as well.

[the remark this answers is not in the document]
Comment (derberg) [reply .github/workflows/upload-release-assets.yml:29]: yup, you basically need 2 `os: windows-latest` matrix configuration, one `extension: x64.exe` and one `extension: x86.exe` if I'm not mistaken

File: docs/installation.md [hunk 3]
```diff
@@ -67,6 +67,11 @@ After downloading the AsyncAPI CLI, install it via the following command:
 sudo installer -pkg asyncapi.pkg -target /
 ```
 
+## Windows 
+For Windows just install the appropriate installer and simply follow the default installation steps to complete the installation process.
+
+Download `aynscapi.x64.exe` for 64-bit Windows and download `asyncapi.x86.exe` for 32-bit Windows.
```
Comment (derberg) [inline docs/installation.md:73]: please make clickable links like https://github.com/asyncapi/cli/releases/latest/download/asyncapi.x64.exe and https://github.com/asyncapi/cli/releases/latest/download/asyncapi.x86.exe

**Comment by Souvikns** [comment]
@derberg how do I test this, do I fork on a different account and try to create a release or something?

**Comment by Souvikns** [comment]
> if you checked that files are really generated and that the pipeline is configured well, the links for sure work.
> 
> I think we are safe to assume it work, as long as above is confirmed

I tested `npm run pack:windows` and `releasePackagesRename.js` script both are working as expected.

**Review by derberg — APPROVED** [review]
LGTM thanks 🚀

**Comment by derberg** [comment]
@Souvikns I don't think the script worked as expected https://github.com/asyncapi/cli/actions/runs/4799227587/jobs/8538571267

**Comment by derberg** [comment]
```
built D:\a\cli\cli\dist\win32\asyncapi-v0.40.0-c3b0028-x86.exe
built D:\a\cli\cli\dist\win32\asyncapi-v0.40.0-c3b0028-x64.exe
```


```
🤔 Pattern 'dist/win32/asyncapi.x64.exe' does not match any files.
👩‍🏭 Creating new GitHub release for tag v$npm_package_version...
🤔 dist/win32/asyncapi.x64.exe not include valid file.
```

**Comment by derberg** [comment]
@Souvikns can you also manually attach windows files you have to https://github.com/asyncapi/cli/releases/tag/v0.40.0

**Comment by derberg** [comment]
@Souvikns last but not least, when you run `npm run get-version --silent` on windows, what do you get?

**Comment by Souvikns** [comment]
> @Souvikns last but not least, when you run `npm run get-version --silent` on windows, what do you get?

I think this does not work as intended
image

**Pull request merged**